x-pack/test/cloud_security_posture_functional/page_objects/findings_page.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import type { FtrProviderContext } from '../ftr_provider_context';
+
+// Defined in EUI
+const getEuiPaginationText = (size: 10 | 25) => `Rows per page: ${size}`;
+const getEuiPaginationRowTestId = (size: 10 | 25) => `tablePagination-${size}-rows`;
+const getEuiPaginationButtonTestId = (index: number) => `pagination-button-${index}`;
+const EUI_PAGINATION_BUTTON = 'tablePaginationPopoverButton';
+
+// Defined in CSP plugin
+const FINDINGS_ROUTE = 'cloud_security_posture/findings';
+const FINDINGS_NAME_COLUMN_TESTID = 'tableHeaderCell_rule.name_5';
+const getNameColumnTestId = (index: string) => `findings_table_cell_rule.name_id-${index}`;
+
+// Defined in archive file
+const getNameColumnText = (index: string) => `Rule ${index}`;
+
+// Defined in Security Solution plugin
+const SECURITY_SOLUTION_APP_NAME = 'securitySolution';
+
+export function FindingsPageProvider({ getService, getPageObjects }: FtrProviderContext) {
+  const testSubjects = getService('testSubjects');
+  const PageObjects = getPageObjects(['common']);
+
+  const navigateToFindingsPage = async () => {
+    await PageObjects.common.navigateToUrl(SECURITY_SOLUTION_APP_NAME, FINDINGS_ROUTE, {
+      shouldUseHashForSubUrl: false,
+    });
+  };
+
+  const goToPageIndex = async (index: number) => {
+    await testSubjects.click(getEuiPaginationButtonTestId(index));
+  };
+
+  const assertPageIndex = async (index: number) => {
+    const isSelected = await testSubjects.getAttribute(
+      getEuiPaginationButtonTestId(index),
+      'aria-current'
+    );
+    expect(isSelected).to.be('true');
+  };
+
+  const changePageSize = async (size: 10 | 25) => {
+    await testSubjects.click(EUI_PAGINATION_BUTTON);
+    await testSubjects.click(getEuiPaginationRowTestId(size));
+  };
+
+  const assertPageSize = async (size: 10 | 25) => {
+    const text = await testSubjects.getVisibleText(EUI_PAGINATION_BUTTON);
+    expect(text).to.be(getEuiPaginationText(size));
+  };
+
+  const assertNameColumnCellValue = async (index: string) => {
+    const text = await testSubjects.getVisibleText(getNameColumnTestId(index));
+    expect(text).to.be(getNameColumnText(index));
+  };
+
+  /**
+   * Note: column name values are defined in the archive file
+   */
+  const assertNameColumnSorting = async (direction: 'asc' | 'desc') => {
```

Comment:
the assertion here takes an assumption about the page size that is being displayed as well as the data structure. which is an unclear assumption that contains several risks in it. 

what I would expect to see when doing assertion for column sorting is to read all name values in the table by their order and check them. (simulating how QA engineer would do that assertion manually)

So this is not tightly coupled to the actual data and light changes in data wouldn't break this assertion.

Reply:
that's a very good point. changed this to grab current list of elements and their text content and assert on that based on expected direction.